src/hotspot/os/windows/os_windows.cpp

```diff
@@ -3330,7 +3330,7 @@ char* os::replace_existing_mapping_with_file_mapping(char* base, size_t size, in
 // Multiple threads can race in this code but it's not possible to unmap small sections of
 // virtual space to get requested alignment, like posix-like os's.
 // Windows prevents multiple thread from remapping over each other so this loop is thread-safe.
-static char* map_or_reserve_memory_aligned(size_t size, size_t alignment, int file_desc) {
+static char* map_or_reserve_memory_aligned(size_t size, size_t alignment, int file_desc, MEMFLAGS flag = mtNone) {
```

Comment:
Wouldn't it be better NOT to implement a default value here?

If the point in this fix is to make sure that we are passing the NMT flags, then we should make sure that we review all the callers, and force them to define the flags explicitly.